raiden_contracts/data/source/services/MonitoringService.sol

```diff
@@ -231,15 +231,14 @@ contract MonitoringService is Utils {
             token_network_address
         ));
 
-        // Only allowed to claim, if channel is settled
-        // Channel is settled if it's data has been deleted
-        TokenNetwork.ChannelState channel_state;
-        (, channel_state) = token_network.getChannelInfo(
+        // Only allowed to claim, if channel is out of the settlement period
+        uint256 settle_block_number;
+        (settle_block_number,) = token_network.getChannelInfo(
```

Comment:
This looks tricky.  `settle_block_number` is sometimes 0, and sometimes the **length** of the settlement period.  I think both `channel_state` and `settle_block_number` are necessary.

Reply:
Makes sense, we should at least check that the channel_state is closed.

Reply:
That should work.